tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -560,11 +560,14 @@ static void ExtendDestinationModel(std::span<const RFieldDescriptor *> newFields
    changeset.fAddedFields.reserve(newFields.size());
    for (const auto *fieldDesc : newFields) {
       auto field = fieldDesc->CreateField(*mergeData.fSrcDescriptor);
-      if (fieldDesc->IsProjectedField())
-         changeset.fAddedProjectedFields.emplace_back(field.get());
-      else
-         changeset.fAddedFields.emplace_back(field.get());
-      changeset.fModel.AddField(std::move(field));
+      if (fieldDesc->IsProjectedField()) {
+         const auto sourceId = fieldDesc->GetProjectionSourceId();
+         const auto &sourceField = mergeData.fSrcDescriptor->GetFieldDescriptor(sourceId);
+         changeset.AddProjectedField(std::move(field),
+                                     [dstName = sourceField.GetFieldName()](const std::string &) { return dstName; });
```

Comment:
I think this is insufficient for field hierarchies with subfields. This function maps all projected fields to the same source field (I was a bit confused by `dstName`, should the capture be called `srcName`?). I think you have to map the entire field hierarchy as we do in `RPageSinkBuf`. There we use an `Internal` function to add a `FieldMap_t` directly, probably as a short-cut to not create a map of field names first to use in a lambda...